src/rust/engine/fs/store/src/remote/base_opendal.rs

```diff
@@ -178,12 +178,7 @@ impl ByteStoreProvider for Provider {
 
     let path = self.path(digest.hash);
 
-    let mut writer = match self
-      .operator
-      .writer_with(&path)
-      .content_length(digest.size_bytes as u64)
```

Comment:
Was this a workaround that is no longer necessary post https://github.com/apache/incubator-opendal/pull/3163, or did this end up in the wrong commit?

Reply:
Coincidence, unrelated to the fixes I made: the `content_length` method was removed in 0.40.0 (https://github.com/apache/incubator-opendal/pull/3044), and the upgrade to a recent HEAD here goes past that.